compiler/testData/diagnostics/tests/operatorsOverloading/assignOperatorOverload.kt

```diff
@@ -0,0 +1,76 @@
+// FIR_IDENTICAL
+// !DIAGNOSTICS: -UNUSED_PARAMETER
+// !LANGUAGE:+AssignOperatorOverloadForJvm
+
+import kotlin.reflect.KProperty
+
+class A
+class B
+operator fun B.assign(a: String) {
+}
+
+class C {
+    fun assign(a: A) {
+    }
+}
+
+operator fun C.assign(a: String) {
+}
+
+operator fun C.assign(a: Int) {
+}
+
+operator fun Int.assign(a: String) {
+}
+
+/**
+ * Test that if return type is not Unit then method has an error
+ */
+<!INAPPLICABLE_OPERATOR_MODIFIER!>operator<!> fun String.assign(a: String): String {
```

Comment:
It would be great to check that error messages are correct in this file (you could use `!RENDER_DIAGNOSTICS_FULL_TEXT`).

Reply:
Used `!RENDER_DIAGNOSTICS_FULL_TEXT`